core/include/seeding/detail/doublet.hpp

```diff
@@ -12,7 +12,16 @@
 
 namespace traccc {
 
-/// Header: unsigned int for the number of doublets per spacepoint bin
+/// Header: the number of doublets per spacepoint bin
+struct doublet_per_bin {
+    unsigned int n_doublets = 0;
+
+    TRACCC_HOST_DEVICE
+    unsigned int get_ref_num() const { return n_doublets; }
+
+    TRACCC_HOST
```

Comment:
Why didn't you use `TRACCC_HOST_DEVICE` here? Even if this function is not used in device code for the moment, it is absolutely fit to be used there. So we might as well allow client code to use it there, if it so desires.

Reply:
Hmm right. It's changed to TRACCC_HOST_DEVICE